src/mpyl/steps/deploy/k8s/__init__.py

```diff
@@ -219,13 +184,23 @@ def deploy_helm_chart(  # pylint: disable=too-many-locals
 ) -> Output:
     run_properties = step_input.run_properties
     project = step_input.project
+    deployment_config = DeployConfig.from_config(run_properties.config)
 
-    deploy_config = DeployConfig.from_config(run_properties.config)
-
-    action = deploy_config.action.value
+    action = deployment_config.action.value
     if action == DeployAction.KUBERNETES_MANIFEST.value:  # pylint: disable=no-member
-        path = Path(project.root_path, deploy_config.output_path)
-        file_path = write_manifest(release_name, path, chart)
+        argo_folder_name = __get_argo_folder_name(target=target)
+        namespace = get_namespace(
+            run_properties=run_properties, project=step_input.project
+        )
+        path = Path(
+            project.root_path,
+            deployment_config.output_path,
+            "k8s-manifests",
+            project.name,
+            argo_folder_name,
+            namespace,
+        )
+        file_path = write_manifest(path, chart)
```

Comment:
Ik think I would find it cleaner to leave the argo specifics out of here and make that a concern of the `push` logic and the file path mappers. That way the logic here can stay simple and generic.
It also gives more flexibility to the end-user to choose different path structures if they'd want to.

Reply:
I tried to implement this, and got quite far.. but I don't see a way to get the project name (the name from the `project.yml`, not from the project path) outside of the actual step. Maybe you or Daniel see a way, feel free to comment or implement in that case.

Reply:
Why does the name of the project path not suffice? You _could_ load the project in the path transformer I guess?